lib/wrap.js

```diff
@@ -9,16 +9,16 @@ const { relay, send } = require('./ipc');
 const resolveMain = require('./resolve-main');
 const suppressExperimentalWarnings = require('./suppress-experimental-warnings');
 
-// Experimental warnings need to be suppressed in worker threads as well, since
-// their process inherits the Node arguments from the main thread.
 suppressExperimentalWarnings(process);
 
-// When using worker threads, each thread appears to require this file through
-// the shared Node arguments (--require), so filter them out here and only run
-// on the main thread.
-if (!isMainThread) return;
+if (isMainThread) {
+  // When using worker threads, each thread inherits execArgv and re-evaulates
+  // --require scripts. Worker threads do not inherit argv, so we copy argv[1]
+  // to the environment for those processes to use.
+  process.env.NODE_DEV_SCRIPT = process.argv[1];
+}
 
-const script = process.argv[1];
+const script = process.env.NODE_DEV_SCRIPT;
```

Comment:
The standard use case does not involve an environment variable called `NODE_DEV_SCRIPT`

We should not be populating an environment variable on every run.

Reply:
How do you suggest we do this instead? worker threads do not inherit argv, and I think polluting the worker data sounds worse that adding an env variable. There is also already a precedent of setting `NODE_DEV_PRELOAD`!

Reply:
I just discovered and refactored this to use `getEnvironmentData`/`setEnvironmentData` which feels like it might be more in line with what you're looking for. I also have #284 which gets rid of the other existing env var as well!